pkg/cmd/cli/cmd/rollback.go

```diff
@@ -0,0 +1,122 @@
+
+  Perform the rollback manually by piping the JSON of the new config back to %[1]s:
+
+  $ %[1]s %[2]s deployment-1 --output=json | %[1]s update deploymentConfigs deployment -f -
+`
+
+func NewCmdRollback(parentName string, name string, f *Factory, out io.Writer) *cobra.Command {
+	rollback := &deployapi.DeploymentConfigRollback{
+		Spec: deployapi.DeploymentConfigRollbackSpec{
+			IncludeTemplate: true,
+		},
+	}
+
+	cmd := &cobra.Command{
+		Use:   fmt.Sprintf("%s <from-deployment>", name),
+		Short: "Revert part of an application back to a previous deployment.",
+		Long:  fmt.Sprintf(rollbackLongDesc, parentName, name),
+		Run: func(cmd *cobra.Command, args []string) {
+			if len(args) == 0 || len(args[0]) == 0 {
+				usageError(cmd, "A deployment name is required.")
+			}
+
+			rollback.Spec.From.Name = args[0]
+
+			outputFormat := kcmd.GetFlagString(cmd, "output")
+			outputTemplate := kcmd.GetFlagString(cmd, "template")
+			dryRun := kcmd.GetFlagBool(cmd, "dry-run")
+
+			osClient, _, err := f.Clients(cmd)
+			checkErr(err)
+
+			namespace := getOriginNamespace(cmd)
+
+			// Generate the rollback config
+			newConfig, err := osClient.DeploymentConfigs(namespace).Rollback(rollback)
+			checkErr(err)
+
+			// If dry-run is specified, describe the rollback and exit
+			if dryRun {
+				describer := &describe.DeploymentConfigDescriber{
+					Client: &rollbackDeploymentDescriberClient{
+						getDeploymentConfig: func(namespace, name string) (*deployapi.DeploymentConfig, error) {
+							return newConfig, nil
+						},
+					},
+				}
+				description, descErr := describer.Describe(newConfig.Namespace, newConfig.Name)
+				checkErr(descErr)
+				out.Write([]byte(description))
+				return
+			}
+
+			// If an output format is specified, display the rollback config JSON and exit
+			// WITHOUT performing a rollback.
+			if len(outputFormat) > 0 {
+				printer, _, perr := kubectl.GetPrinter(outputFormat, outputTemplate)
+				checkErr(perr)
+				printer.PrintObj(newConfig, out)
+				return
+			}
+
+			// Apply the rollback config
+			_, updateErr := osClient.DeploymentConfigs(namespace).Update(newConfig)
+			checkErr(updateErr)
+		},
+	}
+
+	cmd.Flags().BoolVar(&rollback.Spec.IncludeTriggers, "change-triggers", false, "Include the previous deployment's triggers in the rollback")
+	cmd.Flags().BoolVar(&rollback.Spec.IncludeStrategy, "change-strategy", false, "Include the previous deployment's strategy in the rollback")
+	cmd.Flags().BoolVar(&rollback.Spec.IncludeReplicationMeta, "change-scaling-settings", false, "Include the previous deployment's replicationController replica count and selector in the rollback")
+	cmd.Flags().BoolP("dry-run", "d", false, "Instead of performing the rollback, describe what the rollback will look like in human-readable form")
+	cmd.Flags().StringP("output", "o", "", "Instead of performing the rollback, output the rollback deploymenConfig in the specified format (json|yaml|template|templatefile)")
```

Comment:
Don't use "deploymentConfig" when talking to end users.  Use a consistent terminology that clearly identifies what something is in terms a user understands.  There will always be a tension between saying the underlying object and what a user is familiar with, and the users expectations always take priority.

Some of our jargon is deeply technical or arbitrary - all of that eventually has to be rephrased in a way that users can grasp.  Doc and UI has to be recognizable to new users and consistent for experienced users.


> On Jan 30, 2015, at 3:37 PM, Dan Mace <notifications@github.com> wrote:
> 
> In pkg/cmd/cli/cmd/rollback.go:
> 
> > +				checkErr(perr)
> > +				printer.PrintObj(newConfig, out)
> > +				return
> > +			}
> > +
> > +			// Apply the rollback config
> > +			_, updateErr := osClient.DeploymentConfigs(namespace).Update(newConfig)
> > +			checkErr(updateErr)
> > +		},
> > +	}
> > +
> > +	cmd.Flags().BoolVar(&rollback.Spec.IncludeTriggers, "change-triggers", false, "Include the previous deployment's triggers in the rollback")
> > +	cmd.Flags().BoolVar(&rollback.Spec.IncludeStrategy, "change-strategy", false, "Include the previous deployment's strategy in the rollback")
> > +	cmd.Flags().BoolVar(&rollback.Spec.IncludeReplicationMeta, "change-scaling-settings", false, "Include the previous deployment's replicationController replica count and selector in the rollback")
> > +	cmd.Flags().BoolP("dry-run", "d", false, "Instead of performing the rollback, describe what the rollback will look like in human-readable form")
> > +	cmd.Flags().StringP("output", "o", "", "Instead of performing the rollback, output the rollback deploymenConfig in the specified format (json|yaml|template|templatefile)")
> I think I got lost...repeat?
> 
> —
> Reply to this email directly or view it on GitHub.
>